thanosql/resources/_table.py

```diff
@@ -113,28 +113,46 @@ def get(self, name: str, schema: Optional[str] = None) -> Union[Table, dict]:
 
     def update(
         self, name: str, schema: Optional[str] = None, table: Optional[BaseTable] = None
-    ) -> dict:
+    ) -> BaseTable:
         path = f"/{self.tag}/{name}"
         query_params = self._create_input_dict(schema=schema)
         payload = self._create_input_dict(table=table)
 
-        return self.client._request(
+        raw_response = self.client._request(
             method="put", path=path, query_params=query_params, payload=payload
         )
+        
+        if "table" in raw_response:
+            table_adapter = TypeAdapter(BaseTable)
+            parsed_response = table_adapter.validate_python(
+                raw_response["table"]
+            )
+            return parsed_response
```

Comment:
🧐What kind of case does this part handle?

Reply:
The `if "table" in raw_response:` part is to skip cases where an error result is returned, so there's nothing to parse and the response should just be returned as it is.

The
```
table_adapter = TypeAdapter(BaseTable)
            parsed_response = table_adapter.validate_python(
                raw_response["table"]
            )
```
part is Pydantic V2's way of doing parse_obj_as, apparently.. https://docs.pydantic.dev/latest/migration/#introduction-of-typeadapter

Reply:
requests의 status_code는 활용할 수 없나요?

그리고 함수의 기본 리턴이 '정상'인 경우로 로직 순서를 조정해주세요.

```
table_adapter = TypeAdapter(BaseTable) # Can we adjust the scope of this adapter so that it can be used in other functions?

def functions():
    if status_code != 200:
        # error, nothing to parse
        return raw_respnose

    parsed_response = table_adapter.validate_python(raw_response["table"]))
    return parsed_response
```

Reply:
I thought of making a generic util function for methods that require TypeAdapter, but things turned out to be complicated since each method require slightly different treatments (i.e. parsing a list instead of a single object, modifying each member of the list, etc). After looking at the code again, I noticed that only the Table adapter is redundant, so for now I think it is safe to just create a util function for this specific case. If in the future there are methods that require a similar util function, I can modify the util function to cover more general cases.